### PR TITLE
feat(GC): commit after each block GCed

### LINKED ISSUE #2613: GC: delete one block per StoreUpdate
We want to fix #2568 by having an invariant that one StoreUpdate can contain state changes from only one block at a time.
The fix will add an assert, which currently doesn't hold for GC (`clear_data`).

The plan is to have GC commit after every block.

### PR DESCRIPTION
**Pull request by Kouprin**
Fixes #2613.

### CODE REVIEW

File: chain/chain/src/chain.rs [hunk 1]
```diff
@@ -899,13 +932,14 @@ impl Chain {
                     chain_store_update.clear_block_data(block_hash, GCMode::StateSync)?;
```
Comment (mikhailOK) [inline chain/chain/src/chain.rs:932]: here we still have multiple blocks in one commit

File: chain/chain/src/chain.rs [hunk 2]
```diff
@@ -582,8 +582,41 @@ impl Chain {
...
+                blocks_current_height.values().flatten().cloned().collect::<Vec<_>>();
+            for block_hash in blocks_current_height.iter() {
+                let mut current_hash = *block_hash;
+                loop {
+                    // Block `block_hash` is not on the Canonical Chain
+                    // because shorter chain cannot be Canonical one
+                    // and it may be safely deleted
+                    // and all its ancestors while there are no other sibling blocks rely on it.
+                    let mut chain_store_update = self.store.store_update();
+                    if *chain_store_update.get_block_refcount(&current_hash)? == 0 {
```
Comment (mikhailOK) [inline chain/chain/src/chain.rs:602]: here we can do `self.get_block_refcount` and create `chain_store_update` inside

Comment (Kouprin) [reply chain/chain/src/chain.rs]: `get_block_refcount` in ChainStore and ChainStoreUpdate are different

Comment (mikhailOK) [reply chain/chain/src/chain.rs:602]: I see, `ChainStore` turns option into `DBNotFoundErr` and `ChainStoreUpdate` turns `DBNotFoundErr` into 0.
Let's keep `ChainStoreUpdate` for simplicity.

**Pull request merged**